pkg/cache/cache.go

```diff
@@ -274,6 +355,9 @@ func (c *ClusterQueue) deleteWorkload(w *kueue.Workload) {
 		return
 	}
 	c.updateWorkloadUsage(wi, -1)
+	if !apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadPodsReady) {
+		c.WorkloadsWithoutPodsReady.Delete(k)
```

Comment:
what about calling broadcast here?

Reply:
I was thinking about it, it would work. The only reason I haven't done this is performance consideration. For example, when UpdateWorkload is called it would result in two broadcasts - one after delete and one after add. But maybe this is not a concern, especially as now checking the condition is faster, what do you think?

Reply:
we could pass the broadcast function down, but not sure it is the right approach indeed. Keeping as is for now.